source/common/router/config_impl.cc

```diff
@@ -2147,13 +2146,42 @@ PerFilterConfigs::PerFilterConfigs(
     const Protobuf::Map<std::string, ProtobufWkt::Any>& typed_configs,
     const OptionalHttpFilters& optional_http_filters,
     Server::Configuration::ServerFactoryContext& factory_context,
-    ProtobufMessage::ValidationVisitor& validator) {
-  for (const auto& it : typed_configs) {
-    const auto& name = it.first;
-    auto object = createRouteSpecificFilterConfig(name, it.second, optional_http_filters,
-                                                  factory_context, validator);
-    if (object != nullptr) {
-      configs_[name] = std::move(object);
+    ProtobufMessage::ValidationVisitor& validator)
+    : ignore_optional_option_from_hcm_for_route_config_(Runtime::runtimeFeatureEnabled(
+          "envoy.reloadable_features.ignore_optional_option_from_hcm_for_route_config")) {
+
+  static const std::string filter_config_type =
+      envoy::config::route::v3::FilterConfig::default_instance().GetDescriptor()->full_name();
```

Comment:
I think full_name() might already get you a static string? Maybe make this `std::string&` or `absl::string_view` to avoid the extra copy

Reply:
I am not sure the lifetime of full name that returned by the descriptor. So I create a static copy here to avoid possible illegal memory access.

I will check it and if the `full_name()` return static string, I will change it to string_view. :)